tensorzero_internal/src/clickhouse.rs

```diff
@@ -34,32 +36,36 @@ impl ClickHouseConnectionInfo {
     /// However, for tests that directly test ClickHouse behavior, you can directly create the struct.
     pub async fn new(database_url: &str) -> Result<Self, Error> {
         // Add a query string for the database using the URL crate
-        let mut database_url = Url::parse(database_url).map_err(|e| {
+        let parsed_url = Url::parse(database_url).map_err(|e| {
             Error::new(ErrorDetails::Config {
                 message: format!("Invalid ClickHouse database URL: {}", e),
             })
         })?;
 
         #[allow(unused_variables)]
-        let database = validate_clickhouse_url_get_db_name(&database_url)?
+        let database = validate_clickhouse_url_get_db_name(&parsed_url)?
             .unwrap_or_else(|| "default".to_string());
 
         #[cfg(any(feature = "e2e_tests", feature = "batch_tests"))]
         let database = "tensorzero_e2e_tests".to_string();
 
+        // Store the original URL as a SecretString
+        let secret_url = SecretString::from(parsed_url.to_string());
+
         // Although we take the database name from the URL path,
         // we need to set the query string for the database name for the ClickHouse TCP protocol
-        database_url.set_path("");
-        database_url
+        let mut working_url = parsed_url.clone();
+        working_url.set_path("");
+        working_url
             .query_pairs_mut()
             .append_pair("database", &database);
 
         // Set ClickHouse format settings for some error checking on writes
-        set_clickhouse_format_settings(&mut database_url);
+        set_clickhouse_format_settings(&mut working_url);
```

Comment:
Now we're modifying a cloned version i.e. this is a noop. We need to store the modified version. In other words, don't create a new variable `working_url`. You can keep things as `Url` until you're instantiating `Self::Production` below, before which you can convert `database_url` into a secret.